components/TextFields/src/Private/MDCTextInputLayoutCoordinator.m

```diff
@@ -0,0 +1,858 @@
+    _hidesPlaceholderOnInput =
+        [aDecoder decodeBoolForKey:MDCTextInputCoordinatorHidesPlaceholderKey];
+    _leadingUnderlineLabel = [aDecoder decodeObjectForKey:MDCTextInputCoordinatorLeadingLabelKey];
+    _mdc_adjustsFontForContentSizeCategory =
+        [aDecoder decodeBoolForKey:MDCTextInputCoordinatorMDCAdjustsFontsKey];
+    _placeholderLabel = [aDecoder decodeObjectForKey:MDCTextInputCoordinatorPlaceholderLabelKey];
+    _positioningDelegate = [aDecoder decodeObjectForKey:MDCTextInputPositioningDelegateKey];
+    _relativeSuperview =
+        [aDecoder decodeObjectForKey:MDCTextInputCoordinatorCharacterRelativeSuperviewKey];
+    _textColor = [aDecoder decodeObjectForKey:MDCTextInputCoordinatorTextColorKey];
+    _trailingUnderlineLabel = [aDecoder decodeObjectForKey:MDCTextInputCoordinatorTrailingLabelKey];
+    _underlineView = [aDecoder decodeObjectForKey:MDCTextInputCoordinatorUnderlineViewKey];
+  }
+  return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)aCoder {
+  [aCoder encodeObject:self.clearButton forKey:MDCTextInputCoordinatorClearButtonKey];
+  [aCoder encodeObject:self.clearButtonColor forKey:MDCTextInputCoordinatorClearButtonColorKey];
+  [aCoder encodeObject:self.clearButtonImage forKey:MDCTextInputCoordinatorClearButtonImageKey];
+  [aCoder encodeBool:self.hidesPlaceholderOnInput
+              forKey:MDCTextInputCoordinatorHidesPlaceholderKey];
+  [aCoder encodeObject:self.leadingUnderlineLabel forKey:MDCTextInputCoordinatorLeadingLabelKey];
+  [aCoder encodeBool:self.mdc_adjustsFontForContentSizeCategory
+              forKey:MDCTextInputCoordinatorMDCAdjustsFontsKey];
+  [aCoder encodeObject:self.placeholderLabel forKey:MDCTextInputCoordinatorPlaceholderLabelKey];
+  [aCoder encodeObject:self.positioningDelegate forKey:MDCTextInputPositioningDelegateKey];
+  [aCoder encodeObject:self.relativeSuperview
+                forKey:MDCTextInputCoordinatorCharacterRelativeSuperviewKey];
+  [aCoder encodeObject:self.textColor forKey:MDCTextInputCoordinatorTextColorKey];
+  [aCoder encodeObject:self.trailingUnderlineLabel forKey:MDCTextInputCoordinatorTrailingLabelKey];
+  [aCoder encodeObject:self.underlineView forKey:MDCTextInputCoordinatorUnderlineViewKey];
+}
+
+- (instancetype)copyWithZone:(NSZone *)zone {
+  MDCTextInputLayoutCoordinator *copy = [[[self class] alloc] init];
+
+  copy.clearButtonColor = self.clearButtonColor.copy;
+  copy.clearButtonImage = self.clearButtonImage.copy;
+  copy.mdc_adjustsFontForContentSizeCategory = self.mdc_adjustsFontForContentSizeCategory;
+  copy.positioningDelegate = self.positioningDelegate;
+  copy.relativeSuperview = self.relativeSuperview.copy;
+  copy.textColor = self.textColor.copy;
+  copy.underlineView = self.underlineView.copy;
+
+  return copy;
+}
+
+- (void)dealloc {
+  NSNotificationCenter *defaultCenter = [NSNotificationCenter defaultCenter];
+  [defaultCenter removeObserver:self];
+}
+
+- (void)commonInit {
```

Comment:
s/commonInit/MDCTextInputLayoutCoordinatorInit/ to avoid colliding with subclasses.